What are the CachyOS repositories rework (#57)
CachyOS/wiki: src/content/docs/cachyos_repositories/what_are_the_cachyos_repo.md

---
title: CachyOS Repository General Information

> description: Benchmarks and information about the repositories

Reviewer: ```suggestion
description: Benchmarks and information about our optimized repositories
```

---

Our aim to deliver a performance-optimized distribution necessitates recompiling essential Arch Linux packages for generic x86-64,
x86-64-v3, x86-64-v4, and Zen4 architectures.

- x86-64 (Most CPUs) - While lacking AVX2 support found in x86-64-v3, this repository offers optimized packages. LTO and PGO techniques have boosted performance in several packages.
- x86-64-v3- (Intel Haswell and up, AMD Ryzen 1st gen and up) - 5%-20% performance uplift compared to x86-64.
- x86-64-v4 (Intel Rocket Lake, Zen 4, Zen 5) - Delivers substantial performance gains through AVX512 support, depending on the workload.

> - Zen 4 - In addition to the x86-64-v4 instruction set, the following are added:

Reviewer 2: Mention that this will be used for Zen4 and Zen5 CPUs maybe

> ```bash

Reviewer: This kinda fucks up the syntax highlighting. It doesn't look that bad tho so I guess we can leave it

Author: Its to follow Markdown's md040 guidelines.

Author: Although it can be changed for another language in order to have syntax highlighting

Author: I found a solution, pushing...

abm, adx, aes, avx512bf16, avx512bitalg, avx512ifma, avx512vbmi, avx512vbmi2, avx512vnni,

> avx512vpopctndq, clflushopt, clwb, clzero, fsgsbase, gfni, mwaitx, pclmul, pku. prfchw,

Reviewer: ```suggestion
avx512vpopctndq, clflushopt, clwb, clzero, fsgsbase, gfni, mwaitx, pclmul, pku, prfchw,
```

Author: What a little sneaky dot... xD

rpdid, rdrnd, rdseed, sha, sse4a, vaes, vockmulqdq, wbnoinvd, savec, xsaveopt, xsaves
```

> Check out [this](https://en.wikipedia.org/wiki/X86-64#Microarchitecture_levels) page for more information about the instructions used in the different x86_64 variants

Reviewer: ```suggestion
Check out [this](https://en.wikipedia.org/wiki/X86-64#Microarchitecture_levels) page for more information about the instructions used in the different x86_64 micro-architectures
```
